native/src/components/runtime.rs

```diff
@@ -266,12 +281,80 @@ impl Runtime {
         remove_dir_all(extracted).context(CLEANUP_ERROR)?;
 
         info!("Runtime installed!");
+
+        Ok(())
+    }
+
+    #[cfg(target_os = "linux")]
+    pub fn link(&self) -> Result<()> {
+        use std::fs::{copy, create_dir_all};
+        use std::os::unix::fs::symlink;
+
+        use crate::storage::Storage;
+
+        let dirs = ProjectDirs::new()?;
+        let mut storage = Storage::load(&dirs)?;
+
+        if !storage.config.use_linked_runtime {
+            self.uninstall()?;
+        }
+
+        storage.config.use_linked_runtime = true;
+        storage.write(&dirs)?;
+
+        info!("Linking the runtime");
+
+        if Path::new("/usr/lib/firefox/").exists() {
+            for entry in read_dir("/usr/lib/firefox/")?.flatten() {
+                let entry = entry.path();
+                match entry.file_name().expect("Couldn't retrieve a file name").to_str() {
+                    Some("browser") => {
+                        create_dir_all(self.directory.join("browser"))?;
+                        symlink(entry.join("chrome"), self.directory.join("browser/chrome"))?;
+                        symlink(
+                            entry.join("crashreporter-override.ini"),
+                            self.directory.join("browser/crashreporter-override.ini"),
+                        )?;
+                        symlink(entry.join("features"), self.directory.join("browser/features"))?;
+                        symlink(entry.join("omni.ja"), self.directory.join("browser/omni.ja"))?;
+                    }
+                    Some("defaults") => {
+                        create_dir_all(self.directory.join("defaults/pref"))?;
+                        symlink(
+                            entry.join("defaults/pref/channel-prefs.js"),
+                            self.directory.join("defaults/pref/channel-prefs.js"),
+                        )?;
+                    }
+                    Some("firefox-bin") => {
+                        //symlink(self.directory.join("firefox"), self.directory.join("firefox-bin")).ok();
+                    }
+                    Some("firefox") => {
+                        copy(
+                            entry.parent().expect("failed to copy firefox").join("firefox"),
+                            self.directory.join("firefox"),
+                        )?;
+                        // symlink(
+                        //     self.directory.join("firefox"),
+                        //     self.directory.join("firefox-bin"),
+                        // )?;
+                    }
+                    Some(&_) => {
+                        let link = self.directory.join(entry.file_name().unwrap());
+                        symlink(entry, link)?;
+                    }
+                    None => todo!(),
+                }
+            }
```

Comment:
Is it necessary and reliable to separately handle each of those cases? Would it be possible to just symlink everything, except directories where patching needs to be applied (runtime root and `default/pref`)?